CMakeLists.txt

```
@@ -409,8 +351,15 @@ set(Boost_ADDITIONAL_VERSIONS
 # TODO for the brave: transition all mlpack's CMake to 'target-based modern
 # CMake'.  Good luck!  You'll need it.
 set(Boost_NO_BOOST_CMAKE 1)
-find_package(Boost "${BOOST_VERSION}")
-
+if (DISABLE_DOWNLOADS)
+  find_package(Boost "${BOOST_VERSION}" REQUIRED)
+else()
+  find_package(Boost "${BOOST_VERSION}")
+  if (NOT Boost_FOUND)
+    get_deps(https://dl.bintray.com/boostorg/release/1.75.0/source/boost_1_75_0.tar.gz boost boost_1_75_0.tar.gz)
```

```suggestion
    list(GET Boost_ADDITIONAL_VERSIONS 0 Boost_LATEST_VERSION_1)
    string(REPLACE "." "_" Boost_LATEST_VERSION_2 ${Boost_LATEST_VERSION_1})
    get_deps(https://dl.bintray.com/boostorg/release/${Boost_LATEST_VERSION_1}/source/boost_${Boost_LATEST_VERSION_2}.tar.gz boost boost_${Boost_LATEST_VERSION_2}.tar.gz)
```

As latest version of boost is maintained by actions, we can rely on `Boost_ADDITIONAL_VERSIONS`, and Hence we can get rid of updating the version of boost regularly here. Let me know what you think???